### PR TITLE
(feat): easier Kubernetes local development steps

### PR DESCRIPTION
**Pull request by mclarke47**
Signed-off-by: Matthew Clarke <mclarke@spotify.com>

Easier getting started steps for local development of Kubernetes plugin using `kubectl proxy` instead of a service account

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes/src/kubernetes-auth-provider/KubernetesAuthProviders.ts [hunk 1]
```diff
@@ -41,16 +38,23 @@ export class KubernetesAuthProviders implements KubernetesAuthProvidersApi {
     );
     this.kubernetesAuthProviderMap.set(
       'serviceAccount',
-      new ServiceAccountKubernetesAuthProvider(),
+      new ServerSideKubernetesAuthProvider(),
```
Comment (mclarke47) [inline plugins/kubernetes/src/kubernetes-auth-provider/KubernetesAuthProviders.ts:41]: These were all copy-pasted so I deduplicated them

**Comment by mclarke47** [comment]
@freben 🙏  😇

File: plugins/kubernetes-backend/examples/dice-roller/README.md [hunk 2]
```diff
@@ -2,48 +2,35 @@
...
-# Viewing in local Minikube running Backstage locally
+# Viewing in local Kind running Backstage locally
 
 ## Prerequisites
 
 - kubectl installed
-- Minikube installed, with the following addons
-  - metrics-server
-  - ingress
+- Kind installed
```
Comment (Rugvip) [inline plugins/kubernetes-backend/examples/dice-roller/README.md:10]: I'm thinking a link here would be nice tbh 😅

File: plugins/kubernetes/src/kubernetes-auth-provider/ServerSideAuthProvider.ts [hunk 3]
```diff
@@ -17,13 +17,18 @@
 import { KubernetesAuthProvider } from './types';
 import { KubernetesRequestBody } from '@backstage/plugin-kubernetes-common';
 
-export class GoogleServiceAccountAuthProvider
+/**
+ * No-op KubernetesAuthProvider, authorization will be handled in the kubernetes-backend plugin
+ *
+ * @alpha
```
Comment (Rugvip) [inline plugins/kubernetes/src/kubernetes-auth-provider/ServerSideAuthProvider.ts:23]: Is this supposed to be alpha? Seems like something pretty stable to me tbh 😅

File: .changeset/fifty-cars-compare.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kubernetes': minor
+'@backstage/plugin-kubernetes-backend': minor
```
Comment (Rugvip) [inline .changeset/fifty-cars-compare.md:3]: Not actually breaking for the backend right?

I think it's best to highlight the breaking changes in the frontend too, just a short notice of the removed stuff. Might even be best to break into two changesets.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! ❤️

File: plugins/kubernetes/src/kubernetes-auth-provider/ServerSideAuthProvider.ts (hunk 3 above)
Comment (mclarke47) [reply plugins/kubernetes/src/kubernetes-auth-provider/ServerSideAuthProvider.ts:23]: I'm too nervous to cut a v1 😂

File: .changeset/fifty-cars-compare.md (hunk 4 above)
Comment (mclarke47) [reply .changeset/fifty-cars-compare.md:3]: Those remove files should not be used anywhere as they are just no-op objects but I'll add that they have been removed

File: plugins/kubernetes-backend/examples/dice-roller/README.md (hunk 2 above)
Comment (mclarke47) [reply plugins/kubernetes-backend/examples/dice-roller/README.md]: done

**Review by Rugvip — APPROVED** [review]
Nice, let's :shipit: 😁

**Pull request merged**